src/main/java/org/zalando/fahrschein/PersistentCursorManager.java

```diff
@@ -9,35 +9,52 @@
 import java.io.IOException;
 import java.util.Collection;
 
+import static java.lang.String.format;
+
 public class PersistentCursorManager implements CursorManager {
 
     private static final Logger LOG = LoggerFactory.getLogger(PersistentCursorManager.class);
 
     private final JdbcTemplate template;
     private final String consumerName;
+    private final String schemaPrefix;
 
-    public PersistentCursorManager(JdbcTemplate template, String consumerName) {
+    public PersistentCursorManager(final JdbcTemplate template, final String consumerName, final String schema) {
         this.template = template;
         this.consumerName = consumerName;
+        if (schema == null || schema.isEmpty()) {
+            throw new IllegalArgumentException("schema may not be null or empty.");
+        }
+        this.schemaPrefix = schema + ".";
```

Comment:
But the schema can be empty in case it is on the search path. I think you can mark it as nullable and do

    this.schemaPrefix = nullToEmpty(schema) + ".";